src/DocumentFormat.OpenXml/OpenXmlElement.cs

```diff
@@ -197,7 +198,18 @@ internal virtual void ConfigureMetadata(ElementMetadata.Builder builder)
 
         private protected void SetAttribute<TSimpleType>(TSimpleType? value, [CallerMemberName] string propertyName = null!)
             where TSimpleType : OpenXmlSimpleType
-            => ParsedState.Attributes.GetProperty(propertyName).Value = value;
+        {
+            // TODO: Revisit and add corresponding code for TableRow.
+            // Consider adding the following code to the generated code in schemas_openxmlformats_org_wordprocessingml_2006_main.g.cs.
+            // That way, the checks will only have to be performed for Paragraph and TableRow instances.
+            if (this is Paragraph p && propertyName is nameof(Paragraph.ParagraphId) && value is HexBinaryValue hexBinaryValue)
```

Comment:
This is not an ideal place for this. Now `OpenXmlElement` has specific knowledge of these things.

Maybe add a virtual method (for now make it private protected) that can handle this. It may make sense to make this `SetAttribute` virtual but I'm not certain

Reply:
I could not agree more, which is why I added the comment on adding the code to the generated `Paragraph` and `TableRow` classes. Would that be a possibility?

Reply:
You can already do what's needed here (Document at least already has a partial class for this kind of stuff: https://github.com/OfficeDev/Open-XML-SDK/blob/main/src/DocumentFormat.OpenXml/Wordprocessing/Document.cs)

Reply:
The `SetAttribute` method is called by the `ParagraphId` properties of the `Paragraph` and `TableRow` classes. It would be best to change those properties to perform the check that is currently performed by this `SetAttribute` method. However, those properties are **generated**. To my knowledge, they cannot be implemented (again) in a partial class unless they are no longer generated.